src/enumerable.cr

```diff
@@ -1352,6 +1352,9 @@ module Enumerable(T)
     {% if T == String %}
       # optimize for string
       join
+    {% elsif T < Array || T < Iterator %}
```

Comment:
No need for `Iterator` here because `Iterator#+` itself is not defined, so it is inconsistent with `#sum`'s behaviour.

Reply:
I suppose we could also just define `Enumerable#sum` to work with `Iterator` types even if there's no `+` method. But I'm not sure it adds much value over calling `flat_map` explicitly.

Reply:
What's the resolution here?

Reply:
`Iterator` has been removed in a subsequent commit, so this is resolved.